ArchUnitNET/Domain/PlantUml/Export/PlantUmlFileBuilder.cs

```diff
@@ -172,11 +217,319 @@ public PlantUmlFileBuilder WithDependenciesFrom(IEnumerable<Slice> slices, Gener
                 }
             }
 
+            if (!generationOptions.CompactVersion)
+            {
+                RemoveDuplicateDependenciesWhenShowingPackages();
+                ReplaceCirclesWithAppropriateDependencyType();                
+            }
+            RemoveDuplicatedArrowsIfExist();
+
+            _diagram.AddElements(nodeElements.OrderBy(element =>
+                element is PlantUmlNamespace @namespace ? @namespace.Name.Length : -1));
+            _diagram.AddElements(_dependencies);
+
+            return this;
+        }
+
+        public PlantUmlFileBuilder WithDependenciesFromFocusOn(IEnumerable<Slice> slices, string package)
```

Comment:
It feels like this function has a lot in common with `WithDependenciesFrom` above. Surely we can extract some common functions to be used by both?